mutiny-server/src/routes.rs

```diff
@@ -0,0 +1,486 @@
+            fees,
+        };
+        payments.push(payment);
+    }
+
+    Ok(Json(json!(payments)))
+}
+
+pub async fn get_payment(
+    Extension(state): Extension<State>,
+    Path(payment_hash): Path<String>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let hash = Hash::from_str(payment_hash.as_str()).map_err(|_| {
+        (
+            StatusCode::BAD_REQUEST,
+            Json(json!({"error": "invalid payment hash"})),
+        )
+    })?;
+
+    let invoice = state
+        .mutiny_wallet
+        .get_invoice_by_hash(&hash)
+        .await
+        .map_err(handle_mutiny_err)?;
+
+    let pr = invoice.bolt11.as_ref().map(|bolt11| bolt11.to_string());
+    let (is_paid, amount_sats, fees) = get_payment_info(&invoice)?;
+
+    let payment = PaymentResponse {
+        payment_hash: hash.to_string(),
+        preimage: invoice.preimage,
+        description: invoice.description,
+        invoice: pr,
+        incoming: invoice.inbound,
+        is_paid,
+        amount_sats,
+        fees,
+    };
+
+    Ok(Json(json!(payment)))
+}
+
+pub async fn get_balance(
+    Extension(state): Extension<State>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let balance = state
+        .mutiny_wallet
+        .get_balance()
+        .await
+        .map_err(handle_mutiny_err)?;
+    Ok(Json(json!(balance)))
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NodeInfoResponse {
+    node_id: String,
+    channels: Vec<Channel>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Channel {
+    state: String,
+    channel_id: String,
+    balance_sat: u64,
+    inbound_liquidity_sat: u64,
+    capacity_sat: u64,
+    funding_tx_id: Option<String>,
+}
+
+enum ChannelState {
+    Usable,
+    Unusable,
+}
+
+impl fmt::Display for ChannelState {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            ChannelState::Usable => write!(f, "usable"),
+            ChannelState::Unusable => write!(f, "unusable"),
+        }
+    }
+}
+
+pub async fn get_node_info(
+    Extension(state): Extension<State>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let nodes = state
+        .mutiny_wallet
+        .node_manager
+        .list_nodes()
+        .await
+        .map_err(handle_mutiny_err)?;
+
+    let node_pubkey: PublicKey = if !nodes.is_empty() {
+        nodes[0]
+    } else {
+        return Err((
+            StatusCode::INTERNAL_SERVER_ERROR,
+            Json(json!({"error": "unable to get node info"})),
+        ));
+    };
+
+    let channels = state
+        .mutiny_wallet
+        .node_manager
+        .list_channels()
+        .await
+        .map_err(handle_mutiny_err)?;
+
+    let channels = channels
+        .into_iter()
+        .map(|channel| {
+            let state = match channel.is_usable {
+                true => ChannelState::Usable,
+                false => ChannelState::Unusable,
+            };
```

Comment:
instead of all this, can just have `ChannelState` impl `Serialize`, maybe just need to add the serde flag for all lowercase